packages/data/src/components/with-select/index.js

```diff
@@ -56,7 +56,7 @@ const withSelect = ( mapSelectToProps ) => createHigherOrderComponent(
 					ownProps,
 					registry
 				);
-			const mergeProps = useSelect( mapSelect );
+			const mergeProps = useSelect( mapSelect, [ ownProps ] );
```

Comment:
ownProps is a shallow object in general, should we instead try to extract the values from that object as an array (and try to keep the order based on prop names)?

Reply:
> ownProps is a shallow object in general, should we instead try to extract the values from that object as an array (and try to keep the order based on prop names)?

I don't think so, no. For the same reasons as discussed from earlier comments, your suggestion may be considered problematic for how React compares dependencies. But I also don't think it's necessary that we do this. `pure` on the `withSelect` should be enough to keep reference value updates limited to effective changes.

Even if we were to go down this route, I think it would greatly diminish any potential benefit we'd gain over simply running the selector on every render, since trying to derive the dependencies from `ownProps` would itself be a fairly expensive operation.

Reply:
> ownProps is a shallow object in general, should we instead try to extract the values from that object as an array (and try to keep the order based on prop names)?

This is _exactly_ what I had tried in the original `useSelect` pull which led to the React warning I mentioned (which as I noted later is not relevant to this particular pull as I originally thought).